modules/core/src/encoder/SemanticPreEncoder.ts

```diff
@@ -53,75 +53,89 @@ export class SemanticPreEncoder {
 
             } else if (vendor[gvlVendorKey].length === 0) {
 
-              /**
-               * If the vendor does exist, but they haven't declared any
-               * purposes for this legal basis, then we need to see if they can
-               * possibly have the legal basis from their flexible purposes.
-               * This, of course, only matters if it is a globally-scoped string.
-               */
-
-              if (tcModel.isServiceSpecific) {
+              if (
+                gvlVendorKey === 'legIntPurposes' && vendor['purposes'].length === 0 && vendor['legIntPurposes'].length === 0 && vendor['specialPurposes'].length > 0
```

Comment:
this is a great, minimal change. Feels like it needs a unit test. One very minor nit: we already know `vendor['legIntPurposes'].length === 0` from #L54, so that could be omitted, but maybe it's better to repeat it for clarity.

Reply:
Unit tests updated and running

<pre>
TCString
    √ should throw an encoding error if no GVL is set
    √ should unset purposeLegitimateInterests 1 if it is set
    √ should set all vendorsDisclosed in the GVL when isServiceSpecific is false (43ms)
    √ should produce only a publisherTC segment if the encoding option is passed
    √ should add a segment to a model on decode if a model is passed in (52ms)


  5 passing (176ms)
</pre>